### DISCUSSION TITLE
jetson docker deploy failed

### DISCUSSION THREAD

**Discussion by ChungRae** [ultralytics/ultralytics#18299] [Q&A]
I use Dockerfile-jeson-jetpack5 buld docker image succfully, the image name is demo, but creat container failed ,the error msg is :
failed to create link [../../../lib/aarch64-linux-gnu/tegra-egl/nvidia.json /usr/share/glvnd/egl_vendor.d/10_nvidia.json
my commnd is docker exec -ti --gpus all --runtime=nvidia demo:latest

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ChungRae, thank you for reaching out to the Ultralytics community 🚀! We appreciate you sharing your experience. An Ultralytics engineer will assist you shortly.

In the meantime, here are some steps to help troubleshoot your issue and ensure smooth deployment:

If this is a 🐛 Bug Report regarding the Jetson Docker deployment, could you please provide a more detailed [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) and describe your environment setup? Specifically, it would help if you shared:
- The exact commands executed for building the image and attempting to create the container.
- The `Dockerfile` used or any modifications made to it.
- The version of JetPack installed on your Jetson device.
- Confirmation that all dependencies, including the NVIDIA-related drivers and libraries, are properly installed and up-to-date.

We suggest verifying that you are running in a compatible environment, as outlined in our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/).

---

## Upgrade

Ensure your `ultralytics` package, including dependencies, is up-to-date in a [**Python>=3.8**](https://www.python.org/) environment:

```bash
pip install -U ultralytics
```

---

## Verified Environments

YOLO can be run in various verified GPU environments. These include:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **AWS** Deep Learning AMI: [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image** with GPU runtime, as outlined above. <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Check if the CI tests (which validate operation on macOS, Windows, and Ubuntu) are passing. This may give you insights into potential issues.

If you'd like to explore community insights or real-time conversations, join us:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com)
- [Subreddit](https://reddit.com/r/Ultralytics)

Looking forward to your response! 😊

**Reply by ChungRae** [reply]
1.my system version is ubuntu linux/aarch64
2.jetpack is 5.1.1
3.the dockerfile i used is docker/Dockerfile-jetson-jetpack5 , i don't modify it and docker build command is : docker build -f docker/Dockerfile-jetson-jetpack5 -t demo --platform=linux/arm64 .
4.the creat container command is docker run -it --name test --gpus all --runtim=nvidia demo:latest bash
5. the cuda version is 11.4

**Reply by glenn-jocher** [reply]
@ChungRae thank you for providing the details. The issue seems related to the NVIDIA runtime setup or missing symbolic links for NVIDIA libraries. Please confirm that the NVIDIA Container Toolkit is correctly installed and configured by running `docker info | grep -i runtime` to ensure the `nvidia` runtime appears. Additionally, verify that JetPack 5.1.1 and CUDA 11.4 are properly installed with `nvidia-smi`. If the problem persists, consider rebuilding the Docker image and container or refer to the NVIDIA Jetson Docker guide for troubleshooting: https://docs.ultralytics.com/guides/nvidia-jetson/.
